recipe/meta.yaml

```
@@ -37,33 +31,35 @@ requirements:
   build:
     - python
     # run-dep as a workaround for now; upstream package still imports from distutils
-    - setuptools <74  # [py>311]
+    - setuptools <74
+    - {{ compiler('c') }}
     - {{ posix }}bison
     - {{ posix }}flex
     - {{ posix }}patch
     - {{ posix }}sed
     - meson >=0.60.0
     - ninja-base
     - pkg-config
-    - {{ compiler('c') }}
   host:
-    - pkg-config
     - python
     # X11/XCB dependencies through cairo.
     - cairo {{ cairo }}
-    - glib {{ glib }}
+    - glib >=2.82,<2.85
     - libffi {{ libffi }}
     - pthread-stubs 0.3
     - zlib {{ zlib }}  # [win]
-    - setuptools <74
+    - packaging
+    # ../meson.build:29:8: ERROR: $PREFIX/bin/python is missing modules: setuptools
+    - setuptools 72.1.0
```

Based on [this](https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.84.0/meson.build?ref_type=tags#L28) It looks like to me setuptools is only needed on >=3.12. 

```suggestion
    - setuptools 72.1.0  # [py>=312] 
```

---

We should avoid strong pinnings in `host` for Python packages. 
If we will rebuild it or make Python 3.14 buildout it would be better to preserve `setuptools <74` flexibility